vpr/src/route/rr_graph_storage.cpp

```diff
@@ -393,6 +393,24 @@ void t_rr_graph_storage::init_fan_in() {
     }
 }
 
+void t_rr_graph_storage::init_fan_in_list() {
```

Comment:
Because this can be computed using the public interface, just make this a utility class that lives outside of `t_rr_graph_storage` completely.

Reply:
Good idea.

Reply:
I do not think this can be computed using the public interface (unless I am missing something). This method requires the `edge_dest_node_` vector, which is private to the graph storage

Reply:
`for_each_edge` should work here?